### PR TITLE
feat: change Guild.emojis to use collection

### LINKED ISSUE #373: `guild.emojis` should be changed to a collection
At the moment this is an array forcing you to loop over the array every time you want to find an emoji. It would be nice to be able to do guild.emojis.get(emoji.id) which is incredibly fast and efficient.

### PR DESCRIPTION
**Pull request by ayntee**
**Please describe the changes this PR makes and why it should be merged:**
- Convert `guild.emojis` to `Collection<string (emoji ID), Emoji>` from `Emoji[]`

Closes #373

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes

**Semantic versioning classification:**

- [ ] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/api/controllers/guilds.ts [hunk 1]
```diff
@@ -112,11 +113,13 @@ export async function handleInternalGuildEmojisUpdate(data: DiscordPayload) {
   if (!guild) return;
 
   const cachedEmojis = guild.emojis;
-  guild.emojis = payload.emojis;
+  guild.emojis = new Collection(
```
Comment (Skillz4Killz) [inline src/api/controllers/guilds.ts:116]: have u tested this, upon quick review i dont think this will work

Comment (ayntee) [reply src/api/controllers/guilds.ts:116]: Tested what?

**Review by Skillz4Killz — DISMISSED** [review]

**Comment by itohatweb** [comment]
Please update `getEmojis`, `getEmoji` too

**Comment by ayntee** [comment]
> Please update `getEmojis`, `getEmoji` too

Out of scope for this PR

**Comment by itohatweb** [comment]
The checks are failing cause of this

**Comment by ayntee** [comment]
> The checks are failing cause of this

Hmm, makes sense

**Review by itohatweb — APPROVED** [review]
LGTM!

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**